### PR TITLE
Custom notifications and alerting

### PR DESCRIPTION
**Pull request by sfarmar**

### CODE REVIEW

File: Snippets/Snippets_4/Monitoring/MessageFailedHandler.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿namespace MonitoringNotifications.ServiceControl.Notifications
+{
+    using System;
+    using global::ServiceControl.Contracts;
+    using NServiceBus;
+
+	#region MessageFailedHandler 4.0
```
Comment (sfarmar) [inline Snippets/Snippets_4/Monitoring/MessageFailedHandler.cs]: Fixed

File: Snippets/Snippets_5/Monitoring/MessageFailedHandler.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    exceptionMessage,
+                    GetServiceInsightUri(failedMessageId));
+
+                client.PostChatMessage(chatMessage);
+            }
+        }
+		#endregion
+        private string GetServiceInsightUri(string failedMessageId)
+        {
+            //si://localhost:33333/api?Search=26f7b1dc-4ff1-4341-ada4-a3ae01057403
```
Comment (sfarmar) [inline Snippets/Snippets_5/Monitoring/MessageFailedHandler.cs]: This is the uri to build...
Removed...

File: Snippets/Snippets_5/Monitoring/EndpointConfig.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿namespace MonitoringNotifications.ServiceControl.Notifications
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Persistence;
+
+    #region MessageFailedEndpointConfig 5.0
+    public class EndpointConfig : IConfigureThisEndpoint, AsA_Server
```
Comment (sfarmar) [inline Snippets/Snippets_5/Monitoring/EndpointConfig.cs]: I think it is relevant for the event definition and JsonSerializer?

Comment (SimonCropp) [inline Snippets/Snippets_5/Monitoring/EndpointConfig.cs]: how is the JsonSerializer relevant to what you are trying to the point you are trying to illustrate?

Comment (sfarmar) [inline Snippets/Snippets_5/Monitoring/EndpointConfig.cs]: It's a minor pitfall, if overlooked the whole thing will fail... 
Happy to remove it if you feel strongly about it...

**Comment by sfarmar** [comment]
@SimonCropp Added to ServicePulse as this is an extension to the usage of ServicePulse...

File: Snippets/Snippets_5/Monitoring/EndpointConfig.cs (hunk 3 above)
Comment (SimonCropp) [inline Snippets/Snippets_5/Monitoring/EndpointConfig.cs]: >It's a minor pitfall, if overlooked the whole thing will fail...

But you havent actually said that. How is a person meant to know that `JsonSerializer` is important but `RavenDBPersistence` isnt. 

Also the point i was trying to make was that using the host concept of `IConfigureThisEndpoint`  when u just want to illustrate the config api is overkill.

**Comment by sfarmar** [comment]
fixed, moved the link to servicecontrol/plugins

**Comment by sfarmar** [comment]
@SimonCropp ooops, fixed

File: servicepulse/custom_notification_and_alerting_using_servicecontol_events.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+### Subscribing to ServiceControl's Events
+
+ServiceControl publishes MessageFailed event when a message gets to the error queue, let’s see how we can tap in by subscribing to these events and act on them (send an email, pager duty and so on)…
+
+Let's see how we can subscribe to a MessageFailed Event and push a notification into HipChat.
+All it takes is to have an endpoint that subscribes to MessageFailed, and a simple HTTP call to HipChat's API
+
+NOTE: The endpoint will need to match ServiceControl's serieliser: JsonSerializer
```
Comment (SimonCropp) [inline servicepulse/custom_notification_and_alerting_using_servicecontol_events.md]: serieliser

File: servicepulse/custom_notification_and_alerting_using_servicecontol_events.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#### Endpoint config
+<!-- import MessageFailedEndpointConfig -->
+
+#### Custom action example
+<!-- import MessageFailedHandler -->
+
+
+### Alerting on HeartbeatStopped Event
+
+Heartbeats are used to track endpoints health see [this into for more information](/servicepulse/intro-endpoints-heartbeats#active-vs-inactive-endpoints)
```
Comment (SimonCropp) [inline servicepulse/custom_notification_and_alerting_using_servicecontol_events.md]: u need a `.md` on that link

File: Snippets/Snippets_4/Monitoring/SCCluterAppSettings.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<!-- startcode SCClusterAppSettings 1.0 -->
```
Comment (SeanFeldman) [inline Snippets/Snippets_4/Monitoring/SCCluterAppSettings.xml]: XML filename should be `SCClusterAppSettings.xml`?

File: servicepulse/custom_notification_and_alerting_using_servicecontol_events.md (hunk 5 above)
Comment (sfarmar) [inline servicepulse/custom_notification_and_alerting_using_servicecontol_events.md]: Fixed

File: servicepulse/custom_notification_and_alerting_using_servicecontol_events.md (hunk 4 above)
Comment (sfarmar) [inline servicepulse/custom_notification_and_alerting_using_servicecontol_events.md]: Fixed

File: Snippets/Snippets_4/Monitoring/SCCluterAppSettings.xml (hunk 6 above)
Comment (sfarmar) [inline Snippets/Snippets_4/Monitoring/SCCluterAppSettings.xml]: Fixed

File: Snippets/Snippets_4/Monitoring/SCClsuterAppSettings.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<!-- startcode SCClusterAppSettings 1.0 -->
```
Comment (SeanFeldman) [inline Snippets/Snippets_4/Monitoring/SCClsuterAppSettings.xml]: `SCClsuterAppSettings.xml` is even better :)

Comment (sfarmar) [inline Snippets/Snippets_4/Monitoring/SCClsuterAppSettings.xml]: LOL, I'll fix it...

**Comment by sfarmar** [comment]
@SimonCropp can I merge this?

**Pull request merged**